api/client/options.proto

```diff
@@ -3,7 +3,17 @@ syntax = "proto3";
 package nighthawk.client;
 
 import "google/protobuf/duration.proto";
+import "validate/validate.proto";
+import "envoy/api/v2/core/base.proto";
 
+
+message RequestOptions {
+  string request_method = 1;
+  repeated envoy.api.v2.core.HeaderValueOption request_headers = 2;
+  oneof oneof_request_size { 
```

Comment:
Why is this a `oneof`? I think you explained this before but I've forgotten.

Reply:
Using `oneof` is bit of a hack that makes merging work well. So here we can have a default config that specifies we should upload a request body with a certain size by default (hypothetically), and one can merge in '0' to override that. 
I couldn't get this to work with a wrapped type.
We discussed something similar over at https://github.com/envoyproxy/envoy-perf/pull/70/files/3a8f22b235b87a8439b51047ee6e86a88ce442bf#diff-b183cebb7f1f309859fc6a3137456da8

Reply:
This PR generally looks good. I'd like to chat tomorrow about how to use wrapped types here, it seems it should be possible.

Reply:
I backed it out of here, we do not need it yet.
To concretely demonstrate the problem, I created https://github.com/envoyproxy/nighthawk/pull/63 for discussion. That demonstrates the problem clearly by failing a test when introducing the typed wrapper.

Reply:
OK, let's continue there.